### PR TITLE
Extend redis subscriptions service collection extensions to use a factory method

### PR DESCRIPTION
**Pull request by glucaci**

### CODE REVIEW

File: src/Core/Subscriptions.Redis/RedisSubscriptionServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -9,17 +9,17 @@ public static class RedisSubscriptionServiceCollectionExtensions
     {
         public static IServiceCollection AddRedisSubscriptionProvider(
```
Comment (michaelstaib) [inline src/Core/Subscriptions.Redis/RedisSubscriptionServiceCollectionExtensions.cs]: We should also preserve the old overloads... the old overloads could use the new ones underneath.

Comment (glucaci) [reply src/Core/Subscriptions.Redis/RedisSubscriptionServiceCollectionExtensions.cs]: done

**Comment by michaelstaib** [comment]
Can you add tests for the extension methods.... then we can merge it.

**Comment by michaelstaib** [comment]
We will deprecate the current api ... so this has become obsolete.

**Pull request closed without merging**